apps/web/utils/actions/organization.ts

```diff
@@ -0,0 +1,53 @@
+"use server";
+
+import { actionClientUser } from "@/utils/actions/safe-action";
+import { createOrganizationBody } from "@/utils/actions/organization.validation";
+import prisma from "@/utils/prisma";
+import { revalidatePath } from "next/cache";
+import { SafeError } from "@/utils/error";
+
+export const createOrganizationAction = actionClientUser
+  .metadata({ name: "createOrganization" })
+  .schema(createOrganizationBody)
+  .action(async ({ ctx: { userId }, parsedInput: { name, slug } }) => {
+    const existingMembership = await prisma.member.findFirst({
+      where: { userId },
+      select: { id: true },
+    });
+
+    if (existingMembership) {
+      throw new SafeError(
+        "You are already a member of an organization. You can only be part of one organization at a time.",
+      );
+    }
+
+    const existingOrganization = await prisma.organization.findUnique({
+      where: { slug },
+      select: { id: true },
+    });
+
+    if (existingOrganization) {
+      throw new SafeError(
+        "An organization with this slug already exists. Please choose a different slug.",
+      );
+    }
+
+    const organization = await prisma.organization.create({
+      data: {
+        name,
+        slug,
+      },
+    });
```

Comment:
we don't want to use this?
https://www.better-auth.com/docs/plugins/organization#create-an-organization